### PR TITLE
Move binaries to jll packages

### PR DESCRIPTION
**Pull request by aviks**
This uses the BinaryBuilder built binaries via jll packages. 

The binding name has changed, but I havent changed the source yet. Hence this will show lots of deprecation errors when testing. I will make that change (in this PR) once the basic strategy is reviewed and accepted. Doing it now will cause very large diffs that will be impossible to review. 

This PR changes the minimum required Julia version to 1.3. Backporting these to Julia 1.0-1.2 is going to be well nigh impossible -- the long chain of library depedencies, particularly for `SDL2_image` makes this infeasible. Of course, this is the most controversial part of this PR. Personally, I feel that is OK -- existing users will continue to use existing versions of this package. 

Superseeds #39. Once this is merged, I will rebase #30 on top of this.

### CODE REVIEW

**Comment by jonathanBieler** [comment]
Awesome, thanks ! I'm fine with the v1.3 restriction.

**Comment by giordano** [comment]
Try now again, there is a new version of `SDL2_jll`.

**Comment by aviks** [comment]
>  I'm fine with the v1.3 restriction and the overall plan. 

Great!

Tests passed locally on my windows machine, but struggling to get it working on Travis. Will keep trying, the new SDL2_jll should help.

**Comment by giordano** [comment]
> Tests passed locally on my windows machine, but struggling to get it working on Travis.

Tests are passing locally also for me on GNU/Linux, but failing on Travis.  There might be some problems with headless tests?

Also, Travis installation of Windows is not great for multimedia packages, it is known for example that FFMPEG cannot possibly work because of missing libraries: https://github.com/JuliaIO/FFMPEG.jl/issues/14.  I'd try first on AppVeyor.

**Comment by giordano** [comment]
> I'd try first on AppVeyor.

https://github.com/aviks/SimpleDirectMediaLayer.jl/pull/1

**Comment by aviks** [comment]
@giordano Mac is still failing with the iconv issue.

**Comment by giordano** [comment]
> Mac is still failing with the iconv issue.

Yes, I saw that, this time is in `SDL2_image`.  I'm on it.

**Comment by giordano** [comment]
@aviks it should be fixed now

**Comment by aviks** [comment]
@jonathanBieler: I think this is ready to merge now, at your pleasure. 

@giordano: As always, couldn't have done it without you. Grazie!

File: test/lib/SDL.jl [hunk 1]
```diff
@@ -23,7 +23,11 @@ win = SDL2.CreateWindow("Hello World!", Int32(100), Int32(100), Int32(300), Int3
 
 renderer = SDL2.CreateRenderer(win, Int32(-1),
              UInt32(SDL2.RENDERER_ACCELERATED | SDL2.RENDERER_PRESENTVSYNC))
-@test renderer != C_NULL
+if get(ENV, "HAS_JOSH_K_SEAL_OF_APPROVAL", "") == "true" && Sys.islinux()
```
Comment (jonathanBieler) [inline test/lib/SDL.jl]: What are those about ?

Comment (giordano) [reply test/lib/SDL.jl]: Well, we couldn't find a way to make tests pass on Linux with Travis, `xvfb-run` doesn't seem to help here. They do pass locally for me though. This is marking the tests as broken only on Linux in Travis

Comment (giordano) [reply test/lib/SDL.jl]: I see that Linux on Travis never actually worked so far? I think this is still an improvement. Note that these tests are expected to pass on macOS and Windows with Travis and AppVeyor.

Comment (aviks) [reply test/lib/SDL.jl]: Yes, these line disable a few tests when running on linux on Travis. The `HAS_JOSH_K..` env setting is a documented way to test for travis. When running locally on linux, these tests are still executed. The previous setting had the linux tests completey disregarded on linux, so I think this is an improvement (even if not ideal)

Comment (jonathanBieler) [reply test/lib/SDL.jl]: Alright, thanks for the explanation.

**Comment by jonathanBieler** [comment]
Thanks for all the work with the binaries dependency, I got a new mac recently and it's so much straightforward to install things than with the previous system.

**Pull request merged**